app/models/facility.rb

```diff
@@ -104,24 +104,12 @@ def facilities
   end
 
   def cohort_analytics(period, prev_periods)
-    query = CohortAnalyticsQuery.new(registered_hypertension_patients)
-    query.patient_counts_by_period(period, prev_periods)
+    query = CohortAnalyticsQuery.new([self], period: period, prev_periods: prev_periods)
```

Comment:
To be consistent, could we pass in just `self` as the "region", and invoke `region.facilities` inside?